tpot/tpot.py

```diff
@@ -485,314 +468,6 @@ def _parse_scoring_docstring(self, scoring_function):
-
-        return self._train_model_and_predict(input_df, LogisticRegression, C=C,
-            penalty=penalty_selection, dual=dual, random_state=42)
-
-    def _knnc(self, input_df, n_neighbors, weights):
-        """Fits a k-nearest neighbor classifier
-
-        Parameters
-        ----------
-        input_df: pandas.DataFrame {n_samples, n_features+['class', 'group', 'guess']}
-            Input DataFrame for fitting the k-nearest neighbor classifier
-        n_neighbors: int
-            Number of neighbors to use by default for k_neighbors queries; must be a positive value
-        weights: int
-            Selects a value from the list: ['uniform', 'distance']
-
-        Returns
-        -------
-        input_df: pandas.DataFrame {n_samples, n_features+['guess', 'group', 'class', 'SyntheticFeature']}
-            Returns a modified input DataFrame with the guess column updated according to the classifier's predictions.
-            Also adds the classifiers's predictions as a 'SyntheticFeature' column.
-
-        """
-        training_set_size = len(input_df.loc[input_df['group'] == 'training'])
-        n_neighbors = max(min(training_set_size - 1, n_neighbors), 2)
-
-        weights_values = ['uniform', 'distance']
-        weights_selection = weights_values[weights % len(weights_values)]
-
-        return self._train_model_and_predict(input_df, KNeighborsClassifier,
-            n_neighbors=n_neighbors, weights=weights_selection)
-
-    def _gradient_boosting(self, input_df, learning_rate, max_features, min_weight):
-        """Fits the sklearn GradientBoostingClassifier classifier
-
-        Parameters
-        ----------
-        input_df: pandas.DataFrame {n_samples, n_features+['class', 'group', 'guess']}
-            Input DataFrame for fitting the XGBoost classifier
-        learning_rate: float
-            Shrinks the contribution of each tree by learning_rate
-        max_features: float
-            Maximum number of features to use (proportion of total features)
-        min_weight_fraction_leaf: float
-            The minimum weighted fraction of the input samples required to be at a leaf node.
-
-        Returns
-        -------
-        input_df: pandas.DataFrame {n_samples, n_features+['guess', 'group', 'class', 'SyntheticFeature']}
-            Returns a modified input DataFrame with the guess column updated according to the classifier's predictions.
-            Also adds the classifiers's predictions as a 'SyntheticFeature' column.
-
-        """
-        learning_rate = min(1., max(learning_rate, 0.0001))
-        max_features = min(1., max(0., learning_rate))
-        min_weight = min(0.5, max(0., min_weight))
-
-        return self._train_model_and_predict(input_df, GradientBoostingClassifier,
-            learning_rate=learning_rate, n_estimators=500,
-            max_features=max_features, random_state=42, min_weight_fraction_leaf=min_weight)
-
     def _train_model_and_predict(self, input_df, model, **kwargs):
```

Comment:
@teaearlgraycold  I resolved most conflicts (and tried to adopt the new numpy matrices/indices), but I left my original _train_model_and_predict() in because I wanted to see what you think the best way is for passing the clf_eval_func var to the separate classifier operators. In writing this I think I figured it out though -- throw it into a Classifier's _call and keep track of it there.